### DISCUSSION TITLE
Does OpenFAST has provisions to simulate IEC DLC 3.1, 4.1 and 5.1?

### DISCUSSION THREAD

**Discussion by GBA217** [OpenFAST/openfast#2343] [Q&A]
I want to simulate standard IEC startup, normal shutdown and emergency shutdown load cases, but I couldn't find anything about it in the discussions or issues. Is OpenFAST capable of simulating these load cases? If yes, please let me know how.

Thanks

**Comment by jjonkman** [comment]
Dear @gorby217,

There are no limitations preventing OpenFAST from simulating these load cases.  The only limitation would be if the controller you are using for the wind turbine you are simulating includes the necessary control logic for start-up, normal shutdown, and emergency stop situations.  (Some reference controllers only apply to normal operation.)

Best regards,

**Reply by GBA217** [reply]
Thanks for the quick response @jjonkman,
I'm trying to do this to reference IEA-15-240-RWT-Monopile model using user-defined controller from Bladed-style DLL(ROSCO). Does it include the necessary control logic for start-up, normal shutdown, and emergency stop situations?

**Comment by dzalkind** [comment]
We are in the process of adding these DLC cases to the [WEIS tool](https://github.com/WISDEM/WEIS).  [This is the feature branch with the new DLCs](https://github.com/WISDEM/WEIS/tree/DLC_RefactorCaseInputs).  I would expect them to be released by the end of summer.  We are compiling a list of things we'd like ROSCO to do during these cases.  So far, it performs okay without any changes, but I'd change a few minor things like the discontinuity when the startup initiates at 10 sec.  Also, I should set the rotor speed initial condition to 0 rpm.  Here is a time series I generated while developing DLC 3.1 the other day.   Most of the changes have been to ServoDyn, which I have included.

![image](https://github.com/user-attachments/assets/4861a29c-1cb5-495c-858c-2439c192b61f)

[DLC3.1_0_IEA15_0_ServoDyn.dat.txt](https://github.com/user-attachments/files/16364015/DLC3.1_0_IEA15_0_ServoDyn.dat.txt)
